cfme/tests/infrastructure/test_host.py

```diff
@@ -452,30 +450,28 @@ def test_infrastructure_hosts_navigation_after_download(
 
 @test_requirements.infra_hosts
 @pytest.mark.parametrize("num_hosts", [2, 4])
+@pytest.mark.parametrize("hosts_collection", ["provider", "appliance"])
 @pytest.mark.uncollectif(
     lambda provider, num_hosts: provider.one_of(RHEVMProvider, SCVMMProvider) and num_hosts > 2,
     reason=UNCOLLECT_REASON)
 @pytest.mark.meta(blockers=[BZ(1746214, forced_streams=["5.10"])], automates=[1746214, 1784181])
-def test_compare_hosts_from_provider_allhosts(appliance, setup_provider_min_hosts, provider,
-                                              num_hosts):
+def test_infrastructure_hosts_compare(appliance, setup_provider_min_hosts, provider, num_hosts,
+                        hosts_collection):
     """
     Polarion:
         assignee: prichard
         casecomponent: Infra
         caseimportance: high
         initialEstimate: 1/6h
-    Bugzilla:
-        1746214
-
     """
-    ent_slice = slice(0, num_hosts, None)
-    hosts_view = navigate_to(provider.collections.hosts, "All")
-    for h in hosts_view.entities.get_all(slice=ent_slice):
-        h.ensure_checked()
-    hosts_view.toolbar.configuration.item_select('Compare Selected items',
-                                                 handle_alert=True)
-    compare_hosts_view = provider.create_view(ProviderHostsCompareView)
-    assert compare_hosts_view.is_displayed
+    if hosts_collection == 'provider':
+        h_coll = provider.hosts.all()[:num_hosts]
```

Comment:
You can actually do this with `locals()[hosts_collection]` to get the instance that you're looking the collection up on.

`provider.collections.hosts` is equivalent to `provider.hosts`, so the lookup for either becomes the same.

```
locals()[hosts_collection].collections.hosts.all()
```

Reply:
optional/FYI, I see the slicing of the result from `all()` is different, though its not clear why?

Reply:
OK. I see. Neat trick. I updated but used locals()[hosts_collection].collections.hosts to get just the collection, then made the list inside of the call with all()[:num_hosts] .